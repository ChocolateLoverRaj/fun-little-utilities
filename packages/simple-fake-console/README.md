# Simple Fake Console

Create fake consoles for testing purpose

## Usage

### Use it directly

```javascript
const { ConsoleInstance, ActionType, getString } = require('simple-fake-console')
const console = new ConsoleInstance()

console.log('hello', 'world')
console.info(0, 1, 2, 3)
console.error(new Error('foo'))
console.warn('Warning')

// Get all actions
//
// Result: Array<{
//   type: 'log' | 'info' | 'error' | 'warn',
//   data: any[]
// }>
console.getActions()

// Get string from selected types of actions
//
// Result: string
getString({
  console,
  types: [ActionType.Log, ActionType.Info]
})
```

### Use it in test via dependency injection

```typescript
// main.ts

import { Console } from 'simple-fake-console'

export function main (console: Console) {
  console.log('hello', 'world')
}
```

```typescript
// test.ts

import assert from 'assert'
import { ConsoleInstance, ActionType, getString } from 'simple-fake-console'
import { main } from './main'

const fakeConsole = new ConsoleInstance()
main(fakeConsole)

assert.deepStrictEqual(
  fakeConsole.getActions(),
  [{
    type: 'log',
    data: ['hello', 'world']
  }]
)

assert.strictEqual(
  getString({ console, types: [ActionType.Log] }),
  'hello world'
)
```

## License

[MIT](https://git.io/fxKXN) © [Hoàng Văn Khải](https://github.com/KSXGitHub)
