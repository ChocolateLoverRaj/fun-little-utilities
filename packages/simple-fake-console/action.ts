export enum ActionType {
  Log = 'log',
  Info = 'info',
  Error = 'error',
  Warn = 'warn'
}

export type ActionData = ReadonlyArray<any>

export interface Action {
  type: ActionType
  data: ActionData
}

export class ActionInstance implements Action {
  constructor (
    public type: ActionType,
    public data: ActionData
  ) {}
}
