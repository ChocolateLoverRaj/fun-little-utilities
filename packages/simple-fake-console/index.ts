export * from './action'
export * from './console'
export * from './string'
