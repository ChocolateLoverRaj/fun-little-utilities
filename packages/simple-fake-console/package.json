{
  "name": "simple-fake-console",
  "version": "0.1.4",
  "description": "Create fake consoles for testing purpose",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://ksxnodeapps.github.io/fun-little-utilities/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodeapps/fun-little-utilities.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodeapps/fun-little-utilities/issues"
  },
  "keywords": [
    "console",
    "fake",
    "test"
  ],
  "dependencies": {
    "tslib": "^1.10.0",
    "@types/node": "^13.1.6"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
