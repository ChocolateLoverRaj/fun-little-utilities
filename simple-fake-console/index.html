<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>simple-fake-console — References</title>
	<meta name="description" content="">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="assets/js/search.js" data-base=".">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="index.html" class="title">simple-fake-console — References</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-only-exported" />
							<label class="tsd-widget" for="tsd-filter-only-exported">Only exported</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="globals.html">Globals</a>
				</li>
			</ul>
			<h1> simple-fake-console — References</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<div class="tsd-panel tsd-typography">
				<a href="#simple-fake-console" id="simple-fake-console" style="color: inherit; text-decoration: none;">
					<h1>Simple Fake Console</h1>
				</a>
				<p>Create fake consoles for testing purpose</p>
				<a href="#usage" id="usage" style="color: inherit; text-decoration: none;">
					<h2>Usage</h2>
				</a>
				<a href="#use-it-directly" id="use-it-directly" style="color: inherit; text-decoration: none;">
					<h3>Use it directly</h3>
				</a>
				<pre><code class="language-javascript"><span class="hljs-keyword">const</span> { ConsoleInstance, ActionType, getString } = <span class="hljs-built_in">require</span>(<span class="hljs-string">'simple-fake-console'</span>)
<span class="hljs-keyword">const</span> <span class="hljs-built_in">console</span> = <span class="hljs-keyword">new</span> ConsoleInstance()

<span class="hljs-built_in">console</span>.log(<span class="hljs-string">'hello'</span>, <span class="hljs-string">'world'</span>)
<span class="hljs-built_in">console</span>.info(<span class="hljs-number">0</span>, <span class="hljs-number">1</span>, <span class="hljs-number">2</span>, <span class="hljs-number">3</span>)
<span class="hljs-built_in">console</span>.error(<span class="hljs-keyword">new</span> <span class="hljs-built_in">Error</span>(<span class="hljs-string">'foo'</span>))
<span class="hljs-built_in">console</span>.warn(<span class="hljs-string">'Warning'</span>)
<span class="hljs-built_in">console</span>.clear()

<span class="hljs-comment">// Get all actions</span>
<span class="hljs-comment">//</span>
<span class="hljs-comment">// Result: Array&lt;</span>
<span class="hljs-comment">//   {</span>
<span class="hljs-comment">//     type: 'log' | 'info' | 'error' | 'warn',</span>
<span class="hljs-comment">//     data: any[]</span>
<span class="hljs-comment">//   } | {</span>
<span class="hljs-comment">//     type: 'clear'</span>
<span class="hljs-comment">//   }</span>
<span class="hljs-comment">// &gt;</span>
<span class="hljs-built_in">console</span>.getActions()

<span class="hljs-comment">// Get string from selected types of actions</span>
<span class="hljs-comment">//</span>
<span class="hljs-comment">// Result: string</span>
getString({
  <span class="hljs-built_in">console</span>,
  <span class="hljs-attr">types</span>: [ActionType.Log, ActionType.Info]
})</code></pre>
				<a href="#use-it-in-test-via-dependency-injection" id="use-it-in-test-via-dependency-injection" style="color: inherit; text-decoration: none;">
					<h3>Use it in test via dependency injection</h3>
				</a>
				<pre><code class="language-typescript"><span class="hljs-comment">// main.ts</span>

<span class="hljs-keyword">import</span> { Console } <span class="hljs-keyword">from</span> <span class="hljs-string">'simple-fake-console'</span>

<span class="hljs-keyword">export</span> <span class="hljs-function"><span class="hljs-keyword">function</span> <span class="hljs-title">main</span> (<span class="hljs-params"><span class="hljs-built_in">console</span>: Console</span>) </span>{
  <span class="hljs-built_in">console</span>.log(<span class="hljs-string">'hello'</span>, <span class="hljs-string">'world'</span>)
}</code></pre>
				<pre><code class="language-typescript"><span class="hljs-comment">// test.ts</span>

<span class="hljs-keyword">import</span> assert <span class="hljs-keyword">from</span> <span class="hljs-string">'assert'</span>
<span class="hljs-keyword">import</span> { ConsoleInstance, ActionType, getString } <span class="hljs-keyword">from</span> <span class="hljs-string">'simple-fake-console'</span>
<span class="hljs-keyword">import</span> { main } <span class="hljs-keyword">from</span> <span class="hljs-string">'./main'</span>

<span class="hljs-keyword">const</span> fakeConsole = <span class="hljs-keyword">new</span> ConsoleInstance()
main(fakeConsole)

assert.deepStrictEqual(
  fakeConsole.getActions(),
  [{
    <span class="hljs-keyword">type</span>: <span class="hljs-string">'log'</span>,
    data: [<span class="hljs-string">'hello'</span>, <span class="hljs-string">'world'</span>]
  }]
)

assert.strictEqual(
  getString({ <span class="hljs-built_in">console</span>, types: [ActionType.Log] }),
  <span class="hljs-string">'hello world'</span>
)</code></pre>
				<a href="#license" id="license" style="color: inherit; text-decoration: none;">
					<h2>License</h2>
				</a>
				<p><a href="https://git.io/fxKXN">MIT</a> © <a href="https://github.com/KSXGitHub">Hoàng Văn Khải</a></p>
			</div>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals  ">
						<a href="globals.html"><em>Globals</em></a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/action.html">Action</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/actioninstance.html">Action<wbr>Instance</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/getstring.html">get<wbr>String</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
					<li class=" tsd-kind-enum">
						<a href="enums/actiontype.html" class="tsd-kind-icon">Action<wbr>Type</a>
					</li>
					<li class=" tsd-kind-class">
						<a href="classes/consoleinstance.html" class="tsd-kind-icon">Console<wbr>Instance</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/anyclass.html" class="tsd-kind-icon">Any<wbr>Class</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/console.html" class="tsd-kind-icon">Console</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/consoledatabase.html" class="tsd-kind-icon">Console<wbr>Database</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/fnwithdata.html" class="tsd-kind-icon">Fn<wbr>With<wbr>Data</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/fnwithoutdata.html" class="tsd-kind-icon">Fn<wbr>Without<wbr>Data</a>
					</li>
					<li class=" tsd-kind-type-alias">
						<a href="globals.html#actiondata" class="tsd-kind-icon">Action<wbr>Data</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-not-exported">
						<a href="globals.html#symactions" class="tsd-kind-icon">sym<wbr>Actions</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-not-exported">
						<a href="globals.html#symfnwithdata" class="tsd-kind-icon">sym<wbr>FnWith<wbr>Data</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-not-exported">
						<a href="globals.html#symfnwithoutdata" class="tsd-kind-icon">sym<wbr>FnWithout<wbr>Data</a>
					</li>
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-module"><span class="tsd-kind-icon">Module</span></li>
				<li class="tsd-kind-object-literal"><span class="tsd-kind-icon">Object literal</span></li>
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
				<li class="tsd-kind-function tsd-has-type-parameter"><span class="tsd-kind-icon">Function with type parameter</span></li>
				<li class="tsd-kind-index-signature"><span class="tsd-kind-icon">Index signature</span></li>
				<li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li>
				<li class="tsd-kind-type-alias tsd-has-type-parameter"><span class="tsd-kind-icon">Type alias with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-enum"><span class="tsd-kind-icon">Enumeration</span></li>
				<li class="tsd-kind-enum-member"><span class="tsd-kind-icon">Enumeration member</span></li>
				<li class="tsd-kind-property tsd-parent-kind-enum"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-enum"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
				<li class="tsd-kind-interface tsd-has-type-parameter"><span class="tsd-kind-icon">Interface with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-interface"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-interface"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-interface"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
				<li class="tsd-kind-class tsd-has-type-parameter"><span class="tsd-kind-icon">Class with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class"><span class="tsd-kind-icon">Accessor</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-class"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-call-signature tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="assets/js/main.js"></script>
<script>if (location.protocol == 'file:') document.write('<script src="assets/js/search.js"><' + '/script>');</script>
</body>
</html>