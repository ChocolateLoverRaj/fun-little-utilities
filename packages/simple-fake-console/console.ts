import { Action, ActionType, ActionInstance } from './action'

export interface LogFunc {
  (...args: any[]): void
}

export interface Console {
  readonly log: LogFunc
  readonly info: LogFunc
  readonly error: LogFunc
  readonly warn: LogFunc
}

export interface ConsoleDatabase {
  getActions (): ReadonlyArray<Action>
}

export class ConsoleInstance implements Console, ConsoleDatabase {
  private readonly actions = Array<ActionInstance>()

  private readonly logfn = (type: ActionType): LogFunc => (...data) => {
    this.actions.push(new ActionInstance(type, data))
  }

  public readonly getActions = (): ReadonlyArray<Action> => this.actions

  public readonly log = this.logfn(ActionType.Log)
  public readonly info = this.logfn(ActionType.Info)
  public readonly error = this.logfn(ActionType.Error)
  public readonly warn = this.logfn(ActionType.Warn)
}
