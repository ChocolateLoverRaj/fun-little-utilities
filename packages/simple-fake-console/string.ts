import { ActionType } from './action'
import { ConsoleDatabase } from './console'

export function getString (options: getString.Options): string {
  const { console, types, wordSeparator = ' ', lineSeparator = '\n' } = options
  const typeSet = new Set(types)

  return console
    .getActions()
    .filter(x => typeSet.has(x.type))
    .map(x => x.data.join(wordSeparator))
    .join(lineSeparator)

}

export namespace getString {
  export interface Options {
    readonly console: ConsoleDatabase
    readonly types: Iterable<ActionType>
    readonly wordSeparator?: string
    readonly lineSeparator?: string
  }
}
