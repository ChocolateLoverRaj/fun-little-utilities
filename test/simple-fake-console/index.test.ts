import * as xjest from 'extra-jest'
import * as sfc from 'simple-fake-console'

abstract class InitBase {
  public readonly console = new sfc.ConsoleInstance()

  constructor () {
    this.init(this.console)
  }

  protected abstract init (console: sfc.ConsoleInstance): void
}

it('ActionType matches snapshot', () => {
  expect(sfc.ActionType).toMatchSnapshot()
})

describe('Before executing console methods', () => {
  class Init extends InitBase {
    protected init () {
      return undefined
    }
  }

  it('ConsoleInstance::getActions() is empty', () => {
    const { console } = new Init()
    expect(console.getActions()).toEqual([])
  })

  it('getString() matches snapshot', () => {
    const { console } = new Init()

    xjest.snap.safe({
      'log + info': sfc.getString({
        console,
        types: [sfc.ActionType.Log, sfc.ActionType.Info]
      }),
      'error + warn': sfc.getString({
        console,
        types: [sfc.ActionType.Error, sfc.ActionType.Warn]
      })
    })()
  })
})

describe('After executing methods', () => {
  class Init extends InitBase {
    protected init (console: sfc.ConsoleInstance): void {
      console.log('a', 0)
      console.info('b', 1, 2)
      console.error('c', 3, 4, 5)
      console.warn('d', 7, 8, 9, 10)
      console.clear()
      console.log('foo', 'bar')
      console.info('baz')
      console.error('hello', 'world')
      console.warn(123, 456)
      console.clear()
    }
  }

  it('ConsoleInstance::getActions() matches snapshot', () => {
    const { console } = new Init()
    expect(console.getActions()).toMatchSnapshot()
  })

  it('getString() matches snapshot', () => {
    const { console } = new Init()

    xjest.snap.safe({
      'log + info': sfc.getString({
        console,
        types: [sfc.ActionType.Log, sfc.ActionType.Info]
      }),
      'error + warn': sfc.getString({
        console,
        types: [sfc.ActionType.Error, sfc.ActionType.Warn]
      })
    })()
  })
})
