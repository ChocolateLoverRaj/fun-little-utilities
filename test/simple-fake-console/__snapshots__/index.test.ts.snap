// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionType matches snapshot 1`] = `
Object {
  "Clear": "clear",
  "Error": "error",
  "Info": "info",
  "Log": "log",
  "Warn": "warn",
}
`;

exports[`after executing methods ConsoleInstance::getActions() matches snapshot 1`] = `
Array [
  ActionInstance {
    "data": Array [
      "a",
      0,
    ],
    "type": "log",
  },
  ActionInstance {
    "data": Array [
      "b",
      1,
      2,
    ],
    "type": "info",
  },
  ActionInstance {
    "data": Array [
      "c",
      3,
      4,
      5,
    ],
    "type": "error",
  },
  ActionInstance {
    "data": Array [
      "d",
      7,
      8,
      9,
      10,
    ],
    "type": "warn",
  },
  ActionInstance {
    "type": "clear",
  },
  ActionInstance {
    "data": Array [
      "foo",
      "bar",
    ],
    "type": "log",
  },
  ActionInstance {
    "data": Array [
      "baz",
    ],
    "type": "info",
  },
  ActionInstance {
    "data": Array [
      "hello",
      "world",
    ],
    "type": "error",
  },
  ActionInstance {
    "data": Array [
      123,
      456,
    ],
    "type": "warn",
  },
  ActionInstance {
    "type": "clear",
  },
]
`;

exports[`after executing methods getString() matches snapshot 1`] = `
"
error + warn: |-
  c 3 4 5
  d 7 8 9 10
  hello world
  123 456
log + info: |-
  a 0
  b 1 2
  foo bar
  baz

"
`;

exports[`before executing console methods getString() matches snapshot 1`] = `
"
error + warn: ''
log + info: ''

"
`;
